### ISSUE TITLE
[REQUIREMENT][Ruby] CI to run rdoc in ./lib directory after every a successful build

### ISSUE THREAD

**Issue by jessiejuachon** [vmware/singleton#1725]
**Is your feature request related to a problem? Please describe.**
Source code documentation

**Describe the solution you'd like**
Update the CI to run `rdoc -include ./lib/ README.md` under root directory after every successful build. This will generate/update a `doc` directory in root. The `doc` directory will contain auto-generated html files from source code.
![Screen Shot 2022-05-31 at 9 41 43](https://user-images.githubusercontent.com/5719531/171228636-d7ea5ffa-9421-47a0-982a-cb0e68eb6aa0.png)

**Describe alternatives you've considered**
https://ruby.libhunt.com/rdoc-alternatives

**Comment by nannany00** [comment]
@jessiejuachon , I think the document should be generated and published when release build is published, not in CI for each successful build, could you help confirm?

**Comment by jessiejuachon** [comment]
> 

@nannany00 , yes, let's keep this one for Step 3 only. And on second thought, I think that it should be okay to skip steps 1 and 2 because the documentation can be generated from source code anytime.
